oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/AzureUtilities.java

```diff
@@ -74,7 +74,12 @@ public final class AzureUtilities {
     private static final long TOKEN_REFRESHER_DELAY = 1L;
 
     private static final Logger log = LoggerFactory.getLogger(AzureUtilities.class);
-    private static final ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+
+    private static final ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor(runnable -> {
```

Comment:
Making daemon is fine but is there no way we can call close on this?

Reply:
Not closing it properly could cause random issues in tests for example

Reply:
> Making daemon is fine but is there no way we can call close on this?

This [function](https://github.com/apache/jackrabbit-oak/blob/trunk/oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/AzureUtilities.java#L148) is part of the Util class and is being called from several places including other Util classes, closing this is not as straightforward, hence I opted for using daemon threads and adding a shutdown hook to close the executor.

Reply:
This implementation is problematic not only for tests but also for production use. If I understood correctly, every call to `storageCredentialAccessTokenFrom(...)` will start a new refresh tasks. There is no mechanism to close this task when the token is no longer needed, so this can lead to the accumulation of many tasks that will keep on executing until the end of the lifetime of the JVM. This is problematic for several reasons:

- Memory leak of tasks
- Many unnecessary calls to Azure. This can cause issues with API request quotas in Azure.
- The execution of obsolete tasks may interfere with the ones that are still required, blocking them from being executed, which can lead to tokens not being refreshed on time.

A better design would be for the refresh service to be explicitly managed by the caller, requiring the caller to create and stop it when the token is no longer needed.

Reply:
Changed the current logic, moved the code to generate token to a concrete class, where each instance of a class will have a single access token and the same will be refresh periodically. No unnecessary call will be made to azure for token generation each time a blob is requested.

The caller of this class will be responsible for closing the token refresh executor service when the token is no longer needed.